src/server/mod.rs

```diff
@@ -0,0 +1,321 @@
+    /// Hash of startup ledger
+    #[arg(short, long)]
+    root_hash: Option<String>,
+    /// Path to startup blocks directory
+    #[arg(short, long)]
+    startup_dir: Option<PathBuf>,
+    /// Path to directory to watch for new blocks
+    #[arg(short, long)]
+    watch_dir: Option<PathBuf>,
+    /// Path to directory for rocksdb
+    #[arg(short, long)]
+    database_dir: Option<PathBuf>,
+    /// Path to directory for logs (default: stdout)
+    #[arg(short, long)]
+    log_dir: Option<PathBuf>,
+    /// Override an existing db on the path provided by database_dir (default: false)
+    #[arg(long, default_value_t = false)]
+    db_override: bool,
+}
+
+#[derive(Args, Debug, Clone)]
+#[command(author, version, about, long_about = None)]
+pub struct GenesisPath {
+    genesis_ledger: String,
+}
+
+pub struct IndexerConfiguration {
+    genesis_ledger: GenesisRoot,
+    root_hash: BlockHash,
+    startup_dir: PathBuf,
+    watch_dir: PathBuf,
+    database_dir: PathBuf,
+    log_file: Option<PathBuf>,
+}
+
+pub async fn handle_command_line_arguments(
+    args: ServerArgs,
+) -> anyhow::Result<IndexerConfiguration> {
+    debug!("Parsing server args");
+    let root_hash = BlockHash(args.root_hash.unwrap_or({
+        info!("Using default root hash: 3NKeMoncuHab5ScarV5ViyF16cJPT4taWNSaTLS64Dp67wuXigPZ");
+        "3NKeMoncuHab5ScarV5ViyF16cJPT4taWNSaTLS64Dp67wuXigPZ".to_string()
+    }));
+    let startup_dir = args.startup_dir.unwrap();
+    let watch_dir = args.watch_dir.unwrap();
+    let database_dir = args.database_dir.unwrap();
+    let log_dir = args.log_dir;
+
+    info!(
+        "Parsing genesis ledger file at {}",
+        args.genesis_ledger.display()
+    );
+    match ledger::genesis::parse_file(&args.genesis_ledger).await {
+        Err(err) => {
+            error!(
+                reason = "Unable to parse genesis ledger",
+                error = err.to_string(),
+                path = &args.genesis_ledger.display().to_string()
+            );
+            process::exit(100)
+        }
+        Ok(genesis_ledger) => {
+            info!("Genesis ledger parsed successfully!");
+
+            let log_file;
+            if let Some(log_dir) = log_dir {
+                let mut log_number = 0;
+                let mut log_fname =
+                    format!("{}/mina-indexer-log-{}", log_dir.display(), log_number);
+                while tokio::fs::metadata(&log_fname).await.is_ok() {
+                    log_number += 1;
+                    log_fname = format!("{}/mina-indexer-log-{}", log_dir.display(), log_number);
+                }
+                log_file = Some(PathBuf::from(&log_fname));
+            } else {
+                log_file = None;
+            }
+            Ok(IndexerConfiguration {
+                genesis_ledger,
+                root_hash,
+                startup_dir,
+                watch_dir,
+                database_dir,
+                log_file,
+            })
+        }
+    }
+}
+
```

Comment:
tracing requires `#[instrument]` attributes

Reply:
They barf arg debugs to stdout. We need to find a better way. Maybe just use log?

Reply:
i'll take a look